### PR TITLE
Toplevel Menu Visibility and Enabled/Disabled State

### PR DESCRIPTION
**Pull request by shundhammer**
## Problem

@anaselli noticed that enabling or disabling menu items or setting them to hidden does not work for toplevel menus in a menu bar.

## Fix

Added the initial status for toplevel items.

Notice that it is the `menuAction` that needs to be made visible/invisible; when we do that with the QMenu, the menus pop up in the top left corner of the screen: Menus are always hidden until  their corresponding menu action when the user triggers it.

As for enabled/disabled, we really need to do this on the menu, not on the menu action, as @anaselli found out (thx!).

What you see on the screen in the menu bar are the menu _actions_, not the menus; the menus are what pops up later.

### CODE REVIEW

**Comment by anaselli** [comment]
LGTM. 
And testing gives the right output now: 

immagine

**Comment by anaselli** [comment]
There is still a problem though. Enabling menu1 does not work but after doing it twice (e.g. after enabling and disabling again), from that point on works. I'm looking trough mga plugin code to see if i did something different because there works instead

File: src/YQMenuBar.cc [hunk 1]
```diff
@@ -80,6 +80,8 @@ YQMenuBar::rebuildMenuTree()
 
 	QMenu * menu = QMenuBar::addMenu( fromUTF8( item->label() ));
 	item->setUiItem( menu );
+        menu->menuAction()->setEnabled( item->isEnabled() );
```
Comment (anaselli) [inline src/YQMenuBar.cc:83]: This line should be
menu->setEnabled( item->isEnabled() );

Comment (shundhammer) [reply src/YQMenuBar.cc]: Oops... fixed.

Thanks for spotting this!

**Review by anaselli — APPROVED** [review]
LGTM

File: src/YQMenuBar.cc [hunk 2]
```diff
@@ -80,6 +80,8 @@ YQMenuBar::rebuildMenuTree()
 
 	QMenu * menu = QMenuBar::addMenu( fromUTF8( item->label() ));
 	item->setUiItem( menu );
+        menu->setEnabled( item->isEnabled() );
+        menu->menuAction()->setVisible( item->isVisible() );
```
Comment (jreidinger) [inline src/YQMenuBar.cc:84]: NP, maybe github issue but indentation looks inconsistent. ( maybe mix of tabs and spaces? )

File: package/libyui-qt.changes [hunk 3]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Thu Nov 12 15:02:51 UTC 2020 - Stefan Hundhammer <shundhammer@suse.com>
+
+- Honor menu bar toplevel menu enabled/disabled and visibility state
```
Comment (jreidinger) [inline package/libyui-qt.changes:4]: this line is one char too long according to changelog styleguide - https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)#Changelog_section_.28.25changelog.29 ( quote `Limit your maximum line length to 67 characters.` )

**Review by jreidinger — APPROVED** [review]
just NPs

**Comment by mvidner** [comment]
Can you please update https://github.com/yast/yast-ycp-ui-bindings/blob/master/examples/MenuBar-shortcuts-test.rb so that it can demonstrate this bug, and existing yast-ruby-bindings still pass? A draft PR s enough, I can finish it.

File: src/YQMenuBar.cc (hunk 2 above)
Comment (shundhammer) [reply src/YQMenuBar.cc:84]: GitHub could never handle tabs properly.

File: package/libyui-qt.changes (hunk 3 above)
Comment (shundhammer) [reply package/libyui-qt.changes:4]: I really don't give a damn. Whoever came up with that never took our names into account.
And 67 chars; really?!? The 1970s are over.

**Comment by anaselli** [comment]
> Can you please update https://github.com/yast/yast-ycp-ui-bindings/blob/master/examples/MenuBar-shortcuts-test.rb so that it can demonstrate this bug, and existing yast-ruby-bindings still pass? A draft PR s enough, I can finish it.

Has libyui-binding added YMenuBar already?

**Comment by shundhammer** [comment]
> Has libyui-binding added YMenuBar already?

AFAIK no.

**Pull request merged**